jax/_src/lib/xla_bridge.py

```diff
@@ -210,17 +211,23 @@ def register_backend_factory(name, factory, *, priority=0):
 register_backend_factory('tpu_driver', _make_tpu_driver_client,
                          priority=100)
 
+
+def _make_gpu_client(platform_name=None):
+  return xla_client.make_gpu_client(
```

Comment:
You should default to 0 here:

if distributed.global_state.process_id is None:
  node_id = 0
else:
  node_id = distributed.global_state.process_id

Reply:
Isn't it just fewer lines of code to default to `0` in `distributed.py`?

Reply:
Right.. but there are 2 behaviors right now and I am confused about them.

initialize has the default None which raises an error if you don't specify process_id.

But the class variable defaults to 0. Previously, process_id was a required field for GPUs.

Reply:
There are two cases: single-process and distributed:
* In the single-process case, you don't call the distributed init function. The default is 0, which is the correct default.
* In the distributed case, call the init function. You must specify a `process_id` (the init function enforces this), unless you are on cloud TPU in which case it is inferred.